src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/RepositoryIntegrationTest.java

```diff
@@ -318,6 +318,26 @@ void testGetFilesAtCommitInstructorNotInCourseForbidden() throws Exception {
         courseUtilService.updateCourseGroups(TEST_PREFIX, course, "");
     }
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
+    void testGetFilesAtCommitTutorNotInCourseForbidden() throws Exception {
+        prepareRepository();
+        String commitHash = getCommitHash(studentRepository.localGit);
+        courseUtilService.updateCourseGroups("abc", course, "");
+        request.getMap(studentRepoBaseUrl + participation.getId() + "/files-content/" + commitHash, HttpStatus.FORBIDDEN, String.class, String.class);
+        courseUtilService.updateCourseGroups(TEST_PREFIX, course, "");
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "editor1", roles = "EDITOR")
+    void testGetFilesAtCommitEditorNotInCourseForbidden() throws Exception {
+        prepareRepository();
+        String commitHash = getCommitHash(studentRepository.localGit);
+        courseUtilService.updateCourseGroups("abc", course, "");
+        request.getMap(studentRepoBaseUrl + participation.getId() + "/files-content/" + commitHash, HttpStatus.FORBIDDEN, String.class, String.class);
+        courseUtilService.updateCourseGroups(TEST_PREFIX, course, "");
+    }
```

Comment:
This could be done with an extra method, but it is not a must:
```suggestion
    @Test
    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
    void testGetFilesAtCommitTutorNotInCourseForbidden() throws Exception {
        getFilesAtCommitForbidden();
    }

    @Test
    @WithMockUser(username = TEST_PREFIX + "editor1", roles = "EDITOR")
    void testGetFilesAtCommitEditorNotInCourseForbidden() throws Exception {
        getFilesAtCommitForbidden();
    }
    
    void getFilesAtCommitForbidden() throws Exception {
        prepareRepository();
        String commitHash = getCommitHash(studentRepository.localGit);
        courseUtilService.updateCourseGroups("abc", course, "");
        request.getMap(studentRepoBaseUrl + participation.getId() + "/files-content/" + commitHash, HttpStatus.FORBIDDEN, String.class, String.class);
        courseUtilService.updateCourseGroups(TEST_PREFIX, course, "");
    }
```